webapp/src/RepoGit.ts

```diff
@@ -143,27 +144,26 @@ export class RepoGit {
   }
 
   private async writeLangFiles(
-    languages: Record<string, Record<string, string>>,
+    languages: TranslationMap,
     translationsPath: string,
   ): Promise<string[]> {
     const paths: string[] = [];
+    const translateFilenameMap = groupByFilename(languages);
     const result = await Promise.allSettled(
-      Object.keys(languages).map(async (lang) => {
-        const yamlPath = path.join(
-          translationsPath,
-          // TODO: what if language file were yaml not yml?
-          `${lang}.yml`,
-        );
-        const yamlOutput = stringify(unflattenObject(languages[lang]), {
-          doubleQuotedAsJSON: true,
-          singleQuote: true,
+      Object.values(translateFilenameMap).map((filenames) => {
+        Object.entries(filenames).forEach(async ([filename, messages]) => {
+          const yamlPath = path.join(translationsPath, filename);
+          const yamlOutput = stringify(unflattenObject(messages), {
+            doubleQuotedAsJSON: true,
+            singleQuote: true,
+          });
```

Comment:
YamlTranslationAdapter and YamlMessageAdapter constructs the message identifier using both the file system path to the YAML file and, the in-file tree path to the translated message.

Unflattening the message identifier will create in-file tree paths including the segments that were taken from the file system path.

For example, trying this branch on call.zetk.in, produces pull requests that wraps the whole file in a new layer of entries: https://github.com/WULCAN/call.zetk.in/pull/2/files?diff=split&w=1